### PR TITLE
Add `mapObject.mapObjectSkip` for removing object keys

### LINKED ISSUE #36: Add ability to remove a key
Would you accept a pull request which implements the following?

```js
mapObject({one: 1, two: 2}, (key, value) => value === 1 ? [key, value] : null);
//=> {one: 1}
```

The code change will look like this

```diff
-let [newKey, newValue, {shouldRecurse = true} = {}] = mapper(key, value, object);
+let mapResult = mapper(key, value, object)
+if (mapResult === null) continue
+let [newKey, newValue, {shouldRecurse = true} = {}] = mapResult
```

### PR DESCRIPTION
**Pull request by gr2m**
closes #36

I need help with the types please. The setup is different from https://github.com/sindresorhus/p-map/blob/main/index.d.ts, and I wasn't able to figure out how to define the types for the `mapObject.mapObjSkip` symbol

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,3 +64,5 @@ module.exports = (object, mapper, options) => {
 
 	return mapObject(object, mapper, options);
 };
+
+module.exports.mapObjSkip = mapObjSkip;
```
Comment (sindresorhus) [inline index.js:68]:
```suggestion
module.exports.mapObjectSkip = mapObjectSkip;
```

File: index.d.ts [hunk 2]
```diff
@@ -45,6 +45,8 @@ declare namespace mapObject {
 		*/
 		shouldRecurse?: boolean;
 	}
+
+	export type mapObjSkip = symbol
```
Comment (sindresorhus) [inline index.d.ts:49]: This is not correct. I would recommend reading up on `unique symbol` in TypeScript.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
You need to add proper readme docs for the symbol, including description on how it should be used and what it does.

File: index.js (hunk 1 above)
Comment (gr2m) [reply index.js]: done via 5e0367d

**Comment by gr2m** [comment]
> You need to add proper readme docs for the symbol, including description on how it should be used and what it does.

done via a5342ee

File: index.d.ts (hunk 2 above)
Comment (gr2m) [reply index.d.ts]: I saw it implemented at https://github.com/sindresorhus/p-map/blob/main/index.d.ts#L93 and I found the docs for it at https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#unique-symbol. 

But this is not an ES Module yet, so we can't just do this, right?

```ts
export const mapObjectSkip: unique symbol;
```

I think we need to define the `mapObjectSkip` on the `mapObject` namespace, and I tried several approaches, but couldn't get it to work.

How would you do it?

Comment (sindresorhus) [reply index.d.ts]: Just use `declare const` at the top-level to declare the `unique symbol` and then add it to the namespace.

**Comment by sindresorhus** [comment]
The pull request also needs a proper title.

File: index.d.ts (hunk 2 above)
Comment (gr2m) [reply index.d.ts]: done via 1efe1a5

**Comment by gr2m** [comment]
> The pull request also needs a proper title.

I changed it but feel free to change it to your own liking, that's what I do for outside contributions in my own projects.

File: index.d.ts [hunk 3]
```diff
@@ -46,9 +46,28 @@ declare namespace mapObject {
...
+	console.log(result);
+	//=> {one: 1}
+	```
+	*/
+	const mapObjectSkip: typeof UniqueSymbol
 }
 
+// unique symbol cannot be declared in a namespace directly, so we declare it top-level
+// See: https://github.com/sindresorhus/map-obj/pull/38#discussion_r702396878
+declare const UniqueSymbol: unique symbol;
```
Comment (sindresorhus) [inline index.d.ts:69]: It should still have a descriptive name and be camel-cased.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 3 above)
Comment (gr2m) [reply index.d.ts]: camel cased via 358fafd. 

It's only used internally, and for that I think it's perfectly descriptive? Can you rename it yourself to whatever you prefer at this point?

File: index.d.ts [hunk 4]
```diff
@@ -65,6 +67,9 @@ const newObject = mapObject({FOO: true, bAr: {bAz: true}}, (key, value) => [key.
 
 const newObject = mapObject({FOO: true, bAr: {bAz: true}}, (key, value) => [key.toLowerCase(), value], {deep: true});
 //=> {foo: true, bar: {baz: true}}
+
+const newObject = mapObject({one: 1, two: 2}, (key, value) => value === 1 ? [key, value] : mapObject.mapObjectSkip);
+//=> {one: 1}
```
[the remark this answers is not in the document]
Comment (gr2m) [reply index.d.ts:90]: It is

```ts
	/**
	Return this value from a `mapper` function to remove a key from an object.

	```js
	const mapObject = require('map-obj');

	const object = {one: 1, two: 2}
	const mapper = (key, value) => value === 1 ? [key, value] : mapObject.mapObjectSkip
	const result = mapObject(object, mapper);

	console.log(result);
	//=> {one: 1}
	```
	*/
	const mapObjectSkip: typeof UniqueSymbol
```

File: index.d.ts [hunk 5]
```diff
@@ -45,6 +45,8 @@ declare namespace mapObject {
 		*/
 		shouldRecurse?: boolean;
 	}
+
+	export type mapObjectSkip = symbol
```
[the remark this answers is not in the document]
Comment (gr2m) [reply index.d.ts]: Moved to `mapObjectSkip` on the `mapObject` namespace, where it has a doc comment now

File: readme.md [hunk 6]
```diff
@@ -66,6 +69,21 @@ Default: `{}`
 
 Target object to map properties on to.
 
+### mapObject.mapObjectSkip
+
+Return this value from a `mapper` function remove a key from an object.
```
[the remark this answers is not in the document]
Comment (gr2m) [reply readme.md]: addressed in a5e21ab

File: index.d.ts (hunk 3 above)
Comment (sindresorhus) [reply index.d.ts]: Even it's internally, it's good to have descriptive names. You don't call all your variables `const variable1 = …; const variable2 ...`. But it's. I can clean it up later.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**